pkg/cmd/grafana-cli/commands/datamigrations/migrate_secrets.go

```diff
@@ -0,0 +1,202 @@
+package datamigrations
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/cmd/grafana-cli/commands/runner"
+	"github.com/grafana/grafana/pkg/cmd/grafana-cli/logger"
+	"github.com/grafana/grafana/pkg/cmd/grafana-cli/utils"
+	"github.com/grafana/grafana/pkg/services/ngalert/notifier"
+	"github.com/grafana/grafana/pkg/services/secrets"
+	"github.com/grafana/grafana/pkg/services/secrets/manager"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"xorm.io/xorm"
+)
+
+type simpleSecret struct {
+	tableName       string
+	columnName      string
+	isBase64Encoded bool
+}
+
+func (s simpleSecret) migrate(secretsSrv *manager.SecretsService, sess *xorm.Session) error {
+	var rows []struct {
+		Id     int
+		Secret string
+	}
+
+	selectSQL := fmt.Sprintf("SELECT id, %s as secret FROM %s", s.columnName, s.tableName)
+	if err := sess.SQL(selectSQL).Find(&rows); err != nil {
+		return err
+	}
+
+	for _, row := range rows {
+		if len(row.Secret) == 0 {
+			continue
+		}
+
+		var (
+			err     error
+			decoded = []byte(row.Secret)
+		)
+
+		if s.isBase64Encoded {
+			decoded, err = base64.StdEncoding.DecodeString(row.Secret)
+			if err != nil {
+				return err
+			}
+		}
+
+		decrypted, err := secretsSrv.Decrypt(context.Background(), decoded)
+		if err != nil {
+			return err
+		}
+
+		encrypted, err := secretsSrv.EncryptWithDBSession(context.Background(), decrypted, secrets.WithoutScope(), sess)
+		if err != nil {
+			return err
+		}
+
+		updateSQL := fmt.Sprintf("UPDATE %s SET %s = ? WHERE id = ?", s.tableName, s.columnName)
+		if _, err := sess.Exec(updateSQL, encrypted, row.Id); err != nil {
+			return err
+		}
+	}
```

Comment:
Shouldn't you Base64Encode secret back if `isBase64Encoded==true` before updating it in the db?

Reply:
Sure, great catch! 👌🏻